### ISSUE TITLE
VBE-inserted parens not handled in SCP

### ISSUE THREAD

**Issue by bclothier** [rubberduck-vba/Rubberduck#4676]
Splitting into its own issue -- apparently if VBE gets to insert the ()s before we can complete the line, it can mess up the AC.

```
Public Sub Foo|()
```

typing <kbd>Enter</kbd> yields:

```
Public Sub Foo()
    |
End Sub
()
```

_Originally posted by @bclothier in https://github.com/rubberduck-vba/Rubberduck/issue_comments#issuecomment-450202389_

**Comment by retailcoder** [comment]
Thinking this should be fixed by block completion rather than self-closing pairs - SCP doesn't handle the <kbd>Enter</kbd> key. Block completion OTOH will have to handle <kbd>Enter</kbd>, and deal with `Sub` and `Function` and `Property` blocks.

**Comment by bclothier** [comment]
Adding another issue here since this likely needs another unit test to handle this.

In vanilla VBE, if we type:

```
Public Sub Init()|
```

and type <kbd>arrow-up</kbd>, we get:

```
Public Sub Init()
|
End Sub
```

With feature auto-completion enabled, the behavior changes, and we end up with this:
```
|
Public Sub Init()
```

Missing the `End Sub` altogether. Curiously, this does not manifest if we do any further typing (e.g. editing the name). For example:

```
Public Function Init() As Boolean|
```

Typing <kbd>arrow-up</kbd> yields the same behavior as the vanilla VBE.

Mind, I'm not suggesting that we want the vanilla behavior; at minimum I'd expect the `End {procedure}` to be inserted, even if I had went up.

**Comment by retailcoder** [comment]
@bclothier on one hand I'm thinking this needs to be handled by block completion.. On the other it's something that's really a consequence of SCP handling the parens completion here - the line being validated/prettified through SCP is preventing the vanilla-VBE block completion.

Feels like special-casing parameterless procedure blocks in SCP would be an overreach of SCP, and yet while putting that concern with block completion would make sense, it would also leave vanilla completion broken when SCP is enabled without block completion being activated...

...so I'm not sure what the best course of action would be here. Treating it as a SCP edge-case bug and having SCP restore the vanilla block completion seems more and more like the best solution, however tricky that might be to address (block completion will be hijacking this case either way, but SCP can't assume block completion is enabled).
